### PR TITLE
Add token revocation to user authentication

### LINKED ISSUE #9: Ability to revoke user token
Add an endpoint to `/signout` a user, removing them from the database and revoking their token. This will be used to demonstrate Vault plugin development.

### PR DESCRIPTION
**Pull request by joatmon08**
Closes #9.

- Adds a new `/signout` API endpoint
- Adds new table to database to track deleted tokens.

### CODE REVIEW

File: blueprint/stack.hcl [hunk 1]
```diff
@@ -8,7 +8,7 @@ container "db" {
     }
 
     image {
-        name = "hashicorpdemoapp/product-api-db:v0.0.15"
+        name = "hashicorpdemoapp/product-api-db:v0.0.16"
```
Comment (im2nguyen) [inline blueprint/stack.hcl:11]: should this be 0.17?

File: docker_compose/docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '3.7'
 services:
   api:
-    image: "hashicorpdemoapp/product-api:v0.0.12"
+    image: "hashicorpdemoapp/product-api:v0.0.16"
```
Comment (im2nguyen) [inline docker_compose/docker-compose.yml:4]: should this be 0.17?

**Review by im2nguyen — APPROVED** [review]
looks good and all test passed 😄 a nit about versioning for dockerfile

File: docker_compose/docker-compose.yml (hunk 2 above)
Comment (joatmon08) [reply docker_compose/docker-compose.yml:4]: I think we have a chicken or egg problem here - if the functional tests run, they need an existing image. But we don't have an image until we push it to Docker Hub 🤔

**Pull request merged**